docs/csharp/language-reference/keywords/volatile.md

```diff
@@ -9,42 +9,46 @@ helpviewer_keywords:
 ms.assetid: 78089bc7-7b38-4cfd-9e49-87ac036af009
 ---
 # volatile (C# Reference)
+
 The `volatile` keyword indicates that a field might be modified by multiple threads that are executing at the same time. Fields that are declared `volatile` are not subject to compiler optimizations that assume access by a single thread. These restrictions ensure that all threads will observe volatile writes performed by any other thread in the order in which they were performed. There is no guarantee of a single total ordering of volatile writes as seen from all threads of execution.  
   
- The `volatile` modifier is usually used for a field that is accessed by multiple threads without using the [lock](../../../csharp/language-reference/keywords/lock-statement.md) statement to serialize access.  
+ The `volatile` modifier is usually used for a field that is accessed by multiple threads without using the [lock](lock-statement.md) statement to serialize access.  
```

Comment:
Is it worth to mention the [Interlocked](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked?view=netframework-4.7.2) class here?

Reply:
@BillWagner a couple more comments.

"...to serialize access." or "...to synchronize access."?

In general, this paragraph invites to use `volatile` when one needs to synchronize access to a shared field. It sounds like a risky trip. Maybe, this paragraph should be removed at all?

Reply:
I did remove the paragraph. I also added more text from the C# spec in the introductory paragraph.